pandas/tests/reshape/merge/test_multi.py

```diff
@@ -837,3 +837,36 @@ def test_join_multi_wrong_order(self):
         )
 
         tm.assert_frame_equal(result, expected)
+
+
+def test_merge_datetime_index_empty_df():
+
+    midx1 = pd.MultiIndex.from_tuples(
+        [[pd.Timestamp("1950-01-01"), "A"], [pd.Timestamp("1950-01-02"), "B"]],
+        names=["date", "panel"],
+    )
+    frame = DataFrame(
+        data={
+            "data": [1.5, 1.5],
+        },
+        index=midx1,
+    )
+
+    midx2 = pd.MultiIndex.from_tuples([], names=["date", "panel"])
+
+    other = DataFrame(index=midx2, columns=["state"])
+
+    midx3 = pd.MultiIndex.from_tuples(
+        [[pd.Timestamp("1950-01-01"), "A"], [pd.Timestamp("1950-01-02"), "B"]],
+        names=["date", "panel"],
+    )
+
+    expected = DataFrame(
+        data={
+            "data": [1.5, 1.5],
+            "state": [None, None],
+        },
+        index=midx3,
+    )
+    result = frame.merge(other, how="left", on=["date", "panel"])
+    tm.assert_frame_equal(result, expected)
```

Comment:
can u swap and test how ='right'

may want to parametrize to make the cases clear